### PR TITLE
OAuth 2.0 Device Polling Interval - Realms settings/Token Tab +- to change value not working

### LINKED ISSUE #29551: OAuth 2.0 Device Polling Interval - Setting in Realms settings/Token Plus-Minus to change value not working
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

OAuth 2.0 Device Polling Interval - Setting in Realms settings/Token Plus-Minus to change value not working

1. Realms Settings
2. Token Tab
3. OAuth 2.0 Device Polling Interval
4. Use + / - to change the value 
5. Not working value did not change. Button for + / - didn't work

### Version

24.04

### Regression

- [ ] The issue is a regression

### Expected behavior

+1 / -1 for the value in Text field

### Actual behavior

 value in Text field not changed

### How to Reproduce?

1. Realms Settings
2. Token Tab
3. OAuth 2.0 Device Polling Interval
4. Use + / - to change the value 
5. Not working value did not change. Button for + / - didn't work

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ahanamallik1**
The input was taking a String type variable. Fixed it by converting it to a number so that numeric calculations can be done on it. Also, applied a condition for Minus button so that the count is never less than zero since the default value is 0.

Fixes #29551

### CODE REVIEW

**Review by ssilvert — APPROVED** [review]

**Comment by ahanamallik1** [comment]
@ssilvert  thank you for the review and approval.
As I read the approving workflows, it should be approved by maintainer separately as it is first time PR by me.
Whom shall I approach for workflow approvals?

**Review by edewit — DISMISSED** [review]

**Comment by agagancarczyk** [comment]
@ahanamallik1 looks good, but there is some small linting error that will need to be fixed for CI to pass.

File: js/apps/admin-ui/src/realm-settings/TokensTab.tsx [hunk 1]
```diff
@@ -179,8 +179,8 @@ export const RealmSettingsTokensTab = ({
                       id="oAuthDevicePollingInterval"
                       value={field.value}
                       min={0}
-                      onPlus={() => field.onChange(field.value || 0 + 1)}
-                      onMinus={() => field.onChange(field.value || 0 - 1)}
+                      onPlus={() => field.onChange(Number(field?.value) + 1)}
+                      onMinus={() => field.onChange((Number(field?.value) > 0) ? Number(field?.value) - 1 : 0)}
```
Comment (ssilvert) [inline js/apps/admin-ui/src/realm-settings/TokensTab.tsx:183]: This should satisfy the linter.

```suggestion
                      onMinus={() =>
                        field.onChange(
                          Number(field?.value) > 0
                            ? Number(field?.value) - 1
                            : 0
                        )
                      }
```

**Review by ssilvert — CHANGES_REQUESTED** [review]

**Comment by ahanamallik1** [comment]
> @ahanamallik1 looks good, but there is some small linting error that will need to be fixed for CI to pass.
@agagancarczyk
Thank you for your feedback . 
EsLint is already present in node_modules.
I also installed EsLint extension in VSCode. But no success in linting the changed file.

Is there any configuration am I missing ?

File: js/apps/admin-ui/src/realm-settings/TokensTab.tsx [hunk 2]
```diff
@@ -179,8 +179,14 @@ export const RealmSettingsTokensTab = ({
...
                       value={field.value}
                       min={0}
-                      onPlus={() => field.onChange(field.value || 0 + 1)}
-                      onMinus={() => field.onChange(field.value || 0 - 1)}
+                      onPlus={() => field.onChange(Number(field?.value) + 1)}
+                      onMinus={() =>
+                        field.onChange(
+                          Number(field?.value) > 0
+                            ? Number(field?.value) - 1
+                            : 0
```
Comment (ssilvert) [inline js/apps/admin-ui/src/realm-settings/TokensTab.tsx:187]:
```suggestion
                            : 0,
```

**Comment by ssilvert** [comment]
> > @ahanamallik1 looks good, but there is some small linting error that will need to be fixed for CI to pass.
> > @agagancarczyk
> > Thank you for your feedback .
> > EsLint is already present in node_modules.
> > I also installed EsLint extension in VSCode. But no success in linting the changed file.
> 
> Is there any configuration am I missing ?

@ahanamallik1 I think I've fixed it now.

**Comment by ssilvert** [comment]
I messed this up.  I did the commits through web interface and didn't include signoff.  So I'm closing this in favor of https://github.com/keycloak/keycloak/pull/29767

**Pull request closed without merging**